twitter_pr.rs

```diff
@@ -0,0 +1,96 @@
+extern crate reqwest;
+
+use std::sync::mpsc::{Receiver, channel};
+use serde::{Serialize, Deserialize};
+use tokio::stream::StreamExt;
+use tokio::io::{AsyncBufReadExt, StreamReader};
+use serde_json::{Deserializer, StreamDeserializer};
+use std::io::Write;
+
+pub struct TwitterApi {
+    pub bearer_api_token: String
+}
+
+struct RequestStream {
+    receiver: Receiver<bytes::Bytes>
+}
+
+impl std::io::Read for RequestStream {
+    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
+        match self.receiver.recv() {
+            Ok(next) => {
+                let vec = next.to_vec();
+                let vec_len = vec.len();
```

Comment:
Rather than making a copy of vec's len here, you can capture the `usize` return value from `write_all`. Works either way, but can save you a step. Actually, the whole last line could be:

```
buf.write_all(&vec)
```

Since this function returns an `io::Result<usize>`, same as write_all.